packages/shared-lib/src/lib/JobQueueWithClasses.ts

```diff
@@ -89,7 +98,13 @@ export class JobQueueWithClasses {
 			// eslint-disable-next-line no-constant-condition
 			while (true) {
 				const firstIn = this.#queue.shift()
-				if (!firstIn) break
+				if (!firstIn) {
+					// No more jobs on queue.
+					this.#waitForDonePromises.forEach((p) => {
+						p.resolve()
+					})
```

Comment:
```suggestion
					})
					this.#waitForDonePromises.length = 0
```

Should we forget about these promises now, so that this doesn't leak?